packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -345,25 +356,63 @@ export class WorkflowRegistry {
  */
 
 export function selectOutput(
+	wf: Workflow,
 	operator: WorkflowNode<any>,
 	selectedWorkflowOutputId: WorkflowOutput<any>['id']
 ) {
 	operator.outputs.forEach((output) => {
 		output.isSelected = false;
+		output.status = WorkflowPortStatus.NOT_CONNECTED;
 	});
 
 	// Update the Operator state with the selected one
 	const selected = operator.outputs.find((output) => output.id === selectedWorkflowOutputId);
-	if (selected) {
-		selected.isSelected = true;
-		operator.state = Object.assign(operator.state, _.cloneDeep(selected.state));
-		operator.status = selected.operatorStatus ?? OperatorStatus.DEFAULT;
-		operator.active = selected.id;
-	} else {
+	if (!selected) {
 		logger.warn(
 			`Operator Output Id ${selectedWorkflowOutputId} does not exist within ${operator.displayName} Operator ${operator.id}.`
 		);
+		return;
 	}
+
+	selected.isSelected = true;
+	operator.state = Object.assign(operator.state, _.cloneDeep(selected.state));
+	operator.status = selected.operatorStatus ?? OperatorStatus.DEFAULT;
+	operator.active = selected.id;
+
+	// If this output is connected to input port(s), update the input port(s)
+	const hasOutgoingEdges = wf.edges.some((edge) => edge.source === operator.id);
+	if (!hasOutgoingEdges) return;
+
+	selected.status = WorkflowPortStatus.CONNECTED;
+
+	const nodeCache = new Map<WorkflowOutput<any>['id'], WorkflowNode<any>[]>();
+	nodeCache.set(operator.id, []);
+
+	wf.edges.forEach((edge) => {
+		// Update the input port of the direct target node
+		if (edge.source === operator.id) {
+			const targetNode = wf.nodes.find((node) => node.id === edge.target);
+			if (!targetNode) return;
+			// Update the input port of the target node
+			const targetPort = targetNode.inputs.find((port) => port.id === edge.targetPortId);
+			if (!targetPort) return;
+			edge.sourcePortId = selected.id; // Sync edge source port to selected output
+			targetPort.label = selected.label;
+			targetPort.value = selected.value;
+		}
+
+		// Collect node cache
+		if (nodeCache.has(edge.source)) {
+			// Add the node connected to this edge's source
+			nodeCache
+				.get(edge.source)
+				?.push(wf.nodes.find((node) => node.id === edge.target) as WorkflowNode<any>);
+			// Make a new relationship for the nodes attached to this edge's target node
+			if (!nodeCache.has(edge.target)) nodeCache.set(edge.target, []);
+		}
```

Comment:
I think this is buggy, let's say we have this graph:
```
A => B => C => D
```

And my edge list is as follows (you can't rely on order because of deletes, duplicate and whatnot)
- C => D
- B => C
- A => B

If we invalidate on `A`,  your cache starts as `A => []` and you are not going to get any thing for B, C, D because the first two iterations will be skipped